model/task_lifecycle.go

```
@@ -454,6 +469,9 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 			}))
 			return errors.Wrap(err, "error updating build")
 		}
+		if err = tryResetTaskGroup(t); err != nil && !strings.Contains(err.Error(), "not finished") {
+			return errors.Wrap(err, "problem resetting task group")
```

I suspect this should return an error regardless of the error message.